analyzers/tests/SonarAnalyzer.UnitTest/Helpers/KnownTypeTest.cs

```diff
@@ -0,0 +1,95 @@
+    [DataRow("System.Byte[]", true, "Byte", true)]
+    [DataRow("System.Exception[]", false, "Exceptions.Exception[]", true)]
+    [DataRow("System.Action<T>", false, "System.Action", true)]
+    [DataRow("System.Action<T>", true, "System.Action", true, "T")]
+    [DataRow("System.Action<T1>", false, "System.Action", true, "T2")]
+    [DataRow("System.Action<T1, T2>", true, "System.Action", true, "T1", "T2")]
+    [DataRow("System.Action<T1, T2>", false, "System.Action", true, "T2", "T1")]
+    [DataRow("System.Action<T1, T2>", false, "System.Action", true, "T1")]
+    public void Matches_TypeSymbol_CS(string symbolName, bool expectedMatch, string fullTypeName, bool isArray, params string[] genericParameters) =>
+        new KnownType.RegularKnownType(fullTypeName, genericParameters) { IsArray = isArray }
+            .Matches(GetSymbol_CS(symbolName))
+            .Should().Be(expectedMatch);
+
+    [DataTestMethod]
+    [DataRow("System.Collections.Generic.IDictionary(Of TKey, TValue)", false, "System.Collections.Generic.IDictionary", true)]
+    [DataRow("System.Collections.Generic.IDictionary(Of TKey, TValue)", false, "System.Collections.Generic.IDictionary", true, "TKey")]
+    [DataRow("System.Collections.Generic.IDictionary(Of TKey, TValue)", true, "System.Collections.Generic.IDictionary", true, "TKey", "TValue")]
+    [DataRow("System.Collections.Generic.IDictionary(Of TKey, TValue)", false, "System.Collections.Generic.IDictionary", true, "TValue", "TKey")]
+    public void Matches_TypeSymbol_VB(string symbolName, bool expectedMatch, string fullTypeName, bool isArray, params string[] genericParameters) =>
+        new KnownType.RegularKnownType(fullTypeName, genericParameters) { IsArray = isArray }
+            .Matches(GetSymbol_VB(symbolName))
+            .Should().Be(expectedMatch);
+
+    [DataTestMethod]
+    [DataRow("string", true, SpecialType.System_String, "string")]
+    [DataRow("byte", false, SpecialType.System_String, "string")]
+    public void Matches_TypeSymbol_SpecialType_CSharp(string expectedSymbol, bool expectedMatch, SpecialType specialType, string typeName) =>
+        new KnownType.SpecialKnownType(specialType, typeName)
+            .Matches(GetSymbol_CS(expectedSymbol))
+            .Should().Be(expectedMatch);
+
+    [DataTestMethod]
+    [DataRow("string", true, SpecialType.System_String, "string")]
+    [DataRow("byte", false, SpecialType.System_String, "string")]
+    public void Matches_TypeSymbol_SpecialType_VisualBasic(string expectedSymbol, bool expectedMatch, SpecialType specialType, string typeName) =>
+        new KnownType.SpecialKnownType(specialType, typeName)
+            .Matches(GetSymbol_VB(expectedSymbol))
+            .Should().Be(expectedMatch);
+
+    private static ITypeSymbol GetSymbol_CS(string type)
+    {
+        var (tree, model) = TestHelper.CompileCS($@"
+namespace Exceptions {{ public class Exception {{ }} }}
+public class Test<T, T1, T2, T3> {{ public {type} Value; }}");
```

Comment:
Damn, I was hoping to have this on a single line. And then the Exceptions arrived. 😅